sdk/storage/azure_storage_blob/tests/blob_client.rs

```diff
@@ -280,3 +283,210 @@ async fn test_upload_blob_overwrite(ctx: TestContext) -> Result<(), Box<dyn Erro
     container_client.delete_container(None).await?;
     Ok(())
 }
+
+#[recorded::test]
+async fn test_put_block_list(ctx: TestContext) -> Result<(), Box<dyn Error>> {
+    // Recording Setup
+    let recording = ctx.recording();
+    let (options, endpoint) = recorded_test_setup(recording, BlobClientOptions::default()).await;
+    let container_name = recording
+        .random_string::<17>(Some("container"))
+        .to_ascii_lowercase();
+    let blob_name = recording
+        .random_string::<12>(Some("blob"))
+        .to_ascii_lowercase();
+
+    // Act
+    let container_client = ContainerClient::new(
+        &endpoint,
+        container_name.clone(),
+        recording.credential(),
+        Some(options.clone()),
+    )?;
+    container_client.create_container(None).await?;
+
+    let blob_client = BlobClient::new(
+        &endpoint,
+        container_name,
+        blob_name,
+        recording.credential(),
+        Some(options),
+    )?;
+
+    let block_1 = b"AAA";
+    let block_2 = b"BBB";
+    let block_3 = b"CCC";
+
+    blob_client
+        .stage_block(
+            "1",
+            i64::try_from(block_1.len())?,
+            RequestContent::from(block_1.to_vec()),
+            Some(BlobBlockBlobClientStageBlockOptions::default()),
```

Comment:
nit: Is there any difference between passing a default options bag and `None`? Should we just pass `None`? This applies a lot in this file.

Also testing passing default is good though so idk...

Reply:
Yeah this is a tricky one, Heath had mentioned in a previous PR that we really should be urging towards `None` when exposing samples and usage examples to cx, but for testing I think we could go either way (and see value in both).

We could establish a pattern here to keep it consistent, but I don't feel strongly towards either end (default vs. None)

Reply:
Let's use `None` in all cases except when testing options bags where we can include a "default" test.